claripy/backends/backend_z3.py

```diff
@@ -612,8 +612,9 @@ def _add(self, s, c, track=False):
         if track:
             for constraint in c:
                 name = str(hash(constraint))
-                self._hash_to_constraint[name] = constraint
-                s.assert_and_track(constraint, name)
+                if name not in self._hash_to_constraint.keys():
```

Comment:
it'd be more performant to do `if name not in self._hash_to_constraint`, no?

Reply:
Yeap totally agree, my bad. Fixed now